client/components/tinymce/plugins/wpeditimage/plugin.js

```diff
@@ -598,9 +599,9 @@ function wpEditImage( editor ) {
 				}
 				// The paste is somewhere else in the caption DL element.
 				// Prevent pasting in there as it will break the caption.
-				// Make new paragraph under the caption DL and move the caret there.
+				// Replace the caption parent with a new paragraph and move the caret there.
 				p = dom.create( 'p' );
-				dom.insertAfter( p, captionParent );
+				dom.replace( p, captionParent );
```

Comment:
I'm not sure what original case this was meant to handle, but I've asked some folks to double check that this doesn't cause a regression elsewhere.

I've verified that this corrects the select/replace case. I am however still seeing an issue when a media item is deleted:
- Insert an image
- Select the image
- Click the edit icon
- In the media modal, delete the selected item from your library
- Dismiss the modal
- The img node is not removed from the post

If folks publish this, you'll see a broken image like https://gwwartest.wordpress.com/2017/05/08/broken-image/